astropy/stats/tests/test_sigma_clipping.py

```diff
@@ -302,3 +302,44 @@ def test_sigma_clippped_stats_all_masked():
     mask = arr < 20
     result = sigma_clipped_stats(arr, mask=mask)
     assert result == (np.ma.masked, np.ma.masked, np.ma.masked)
+
+
+def test_sigma_clip_masked_data_values():
+    """
+    Test that the data values & type returned by sigma_clip are the same as
+    its input when using masked=True (rather than being upcast to float64 &
+    containing NaNs as in issue #10605) and also that the input data get
+    copied or referenced as appropriate.
+    """
+
+    data = np.array([-2, 5, -5, -6, 20, 14, 1])
+
+    result = sigma_clip(data, sigma=1.5, maxiters=3, axis=None, masked=True,
+                        copy=True)
+
+    assert result.dtype == data.dtype
+    assert np.all(result.data == data)
```

Comment:
```suggestion
    assert_equal(result.data, data)
```

Please use `np.testing.assert_equal` instead (https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_equal.html).  `arr1 == arr2` elementwise comparison is deprecated and it also doesn't catch issues with NaNs (`np.nan == np.nan` returns False) or mismatched shapes, etc.

Reply:
OK. Yes, I had done something closer to that (`assert array_equal()`) in my other PR. This was something I thought about changing then forgot; obviously I don't write enough tests, hehe.